### ISSUE TITLE
update dtypes to better support scalars

### ISSUE THREAD

**Issue by hokiegeek2** [Bears-R-Us/arkouda#1077]
1. remove duplicate entries
2. expand int and uint scalars per feedback from @reuster986

**Comment by pierce314159** [comment]
@hokiegeek2, does this issue still need to be worked?

**Comment by hokiegeek2** [comment]
@pierce314159 yeah, I need to revisit this and rock it out. I will do so ASAP.

**Comment by mhmerrill** [comment]
@hokiegeek2 is this done yet?

**Comment by hokiegeek2** [comment]
@mhmerrill no, I've not had a chance to work on it

**Comment by hokiegeek2** [comment]
@mhmerrill @reuster986 @pierce314159 it looks like adding unsigned ints to ARKOUDA_SUPPORTED_NUMBERS ...

![image](https://user-images.githubusercontent.com/10785153/169051259-d023c876-58cf-41b8-85c3-fc6f35f46df2.png)

....

...causes mypy type checking to fail in Python 3.9 (installed w/ python3-pip in the chapel docker)...

![image](https://user-images.githubusercontent.com/10785153/169056151-eaa9e8d7-e90b-411f-934f-ec6afdca8ec0.png)

![image](https://user-images.githubusercontent.com/10785153/169055140-38d04418-2599-4668-9a89-9169d2da578a.png)

The type checking works fine in 3.7 (gonna upgrade to 3.8 ASAP).

So I need to investigate this further

**Comment by hokiegeek2** [comment]
@reuster986 @mhmerrill 

I fixed the type error at timeclass.py line 474:

![image](https://user-images.githubusercontent.com/10785153/170008138-558dbf86-b9b4-46b3-97d5-76f73904988d.png)

WRT to the other errors, it does not appear that Arkouda supports uint8, uint16, or uint32 pdarrays:

![image](https://user-images.githubusercontent.com/10785153/170023400-1613e984-eaed-4109-85e2-ba4467c45e6c.png)

![image](https://user-images.githubusercontent.com/10785153/170023543-c66e4e1e-2470-44e1-a98c-9224da31522b.png)

![image](https://user-images.githubusercontent.com/10785153/170023727-ad447ab9-ee64-43c7-a40b-2b384254f628.png)

![image](https://user-images.githubusercontent.com/10785153/170024296-bfa86436-e6e1-4dc8-b5d3-486498310592.png)

Consequently, expanding the list of scalars to support as I am doing here doesn't appear to make sense:

![image](https://user-images.githubusercontent.com/10785153/170024026-d87b1bfa-9877-429f-b739-52426e212d28.png)

@reuster986 I may be missing something here, so please lemme know what you think.

**Comment by reuster986** [comment]
@hokiegeek2 you're correct that arkouda does not support pdarrays with dtypes like `np.uint16`, but users should still be able to input values of any np dtype wherever an arkouda function expects a scalar (not a pdarray or dtype). For example, if I call `ak.arange(size)` and `size` is a `np.uint16`, arkouda should not raise an error, but should return an int64 pdarray, same as if `size` were a python `int`. In your example above, the right kind of thing to test would be:

```python
>>> u_nums = ak.randint(low=np.uint8(1), high=np.uint16(100), size=np.uint32(100))
```

instead of `dtype=np.uint8`, which will not work, as you observed. In general, the `dtype` arg is not the right place to test these scalar dtypes. Rather, please identify functions with scalar args and test them by supplying values of various np dtypes.

**Comment by hokiegeek2** [comment]
@reuster986 cool, all makes sense, will proceed accordingly

**Comment by hokiegeek2** [comment]
@reuster986 okay, looking good. I confirmed that this type error is thrown in arkouda master:

![image](https://user-images.githubusercontent.com/10785153/172652686-c0971d34-7399-4c23-802c-cab783ab654e.png)

In my feature branch it works fine:

![image](https://user-images.githubusercontent.com/10785153/172652899-8a1f45a5-8cf6-4418-8f7c-eb08b2e2222f.png)

**Comment by hokiegeek2** [comment]
and another, so I think we stuck the landing on this:

![image](https://user-images.githubusercontent.com/10785153/172655710-ebb359c1-3d35-43b9-90e3-af23c82df782.png)

**Comment by pierce314159** [comment]
This was resolved by PR #1492
